### PR TITLE
pulp_soc: bump to v1.4.0, update i2c & spi

### PR DESCRIPTION
**Pull request by micprog**
Updates to SPI and I2C connected to pulp_soc based on changes in pulpissimo: https://github.com/pulp-platform/pulpissimo/commit/c35bc1574fa5b6011b86b7c292c60c1df6d82082

### CODE REVIEW

**Comment by vogelpi** [comment]
Hi @OttG ,
could you please have a look at this? These are just version upgrades and simple bug fixes taken over from PULPissimo.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/.cached*
+/__pycache__*
+/ips
+/ipstools
+/sim/modelsim*
+/sim/ncompile/*
+!/sim/ncompile/build.mk
+/tests/*/
```
Comment (vogelpi) [inline .gitignore:8]: Nit: add missing newline

**Comment by OttG** [comment]
I have reviewed the code and the changes looks good. Do we have any regression tests for SPI and I2C? @accuminium

**Comment by accuminium** [comment]
> I have reviewed the code and the changes looks good. Do we have any regression tests for SPI and I2C? @accuminium

I think so; https://github.com/pulp-platform/pulp/blob/master/tests/runtime-tests.yaml has a couple of entries that contain `i2c` or `spi`.

**Comment by OttG** [comment]
Sorry for the delay but I had quite a busy pipeline. I checked with those tests we had in the pulp folder but I can't find them to work even in the master. From what I know @adimauro-iis should have been working on peripheral tests. I will contact him and update everyone.

**Comment by vogelpi** [comment]
Thanks @OttG for checking the tests and getting back. It would be great if we could push this forward.

**Pull request merged**